### PR TITLE
Elmer and OpenFOAM on LUMI

### PR DESCRIPTION
**Pull request by rkronberg**
https://csc-guide-preview.rahtiapp.fi/origin/lumi-elmer-openfoam/apps/by_system/#lumi
https://csc-guide-preview.rahtiapp.fi/origin/lumi-elmer-openfoam/apps/openfoam/
https://csc-guide-preview.rahtiapp.fi/origin/lumi-elmer-openfoam/apps/elmer/

### CODE REVIEW

File: docs/apps/elmer.md [hunk 1]
```diff
@@ -17,35 +17,37 @@ directives implemented.
...
+Elmer is licensed under GPL and (as for the `elmersolver.lib`) LGPL, hence free to use for everyone.
 
 ## Usage
 
 Elmer version available can be listed using the command
 
 ```bash
-$ module avail elmer
+module avail elmer
 ```
```
Comment (hdrei) [inline docs/apps/elmer.md:35]:
```suggestion
On LUMI, one needs to first take into use the module files of CSC installed software:
```bash
module use /appl/local/csc/modulefiles/
module avail elmer
```

```

File: docs/apps/elmer.md [hunk 2]
```diff
@@ -97,4 +99,4 @@ It can be advantageous to utilize fewer cores per node as available, which can i
```
Comment (hdrei) [inline docs/apps/elmer.md:99]: ### Example parallel batch script for LUMI

```bash
#!/bin/bash 
#SBATCH --time=06:00:00
#SBATCH --job-name=jobname
#SBATCH --output=%x.%j.out
#SBATCH --error=%x.%j.err
#SBATCH --partition=small
#SBATCH --account=<project>
#SBATCH --nodes=4
#SBATCH --ntasks=128

module load elmer/latest
# make sure the SIF is in the start-info
echo myrun.sif > ELMERSOLVER_STARTINFO
echo "starting Elmer simulation with SIF file"
cat ELMERSOLVER_STARTINFO
srun ElmerSolver
echo "done"
```

**Review by hdrei — CHANGES_REQUESTED** [review]
@leopekkas (or @tzwinger ) should check the 'Example parallel batch script for LUMI' before merging

File: docs/apps/elmer.md (hunk 2 above)
Comment (leopekkas) [reply docs/apps/elmer.md:102]: This is a good example of a run where we undersubscribe a node, similarly to the example above with Mahti. If we don't want to explicitly undersubscribe the nodes in this example, then we can switch `--ntasks=128` to `--ntasks-per-node=128` here.

File: docs/apps/elmer.md [hunk 3]
```diff
@@ -17,84 +17,115 @@ directives implemented.
...
+    #!/bin/bash 
+    #SBATCH --time=01:00:00
+    #SBATCH --job-name=jobname
+    #SBATCH --output=%x.%j.out
+    #SBATCH --error=%x.%j.err
+    #SBATCH --partition=small
+    #SBATCH --account=<project>
+    #SBATCH --nodes=4
+    #SBATCH --ntasks=128
+    module use /appl/local/csc/modulefiles/
```
Comment (hdrei) [inline docs/apps/elmer.md:115]:
```suggestion

    module use /appl/local/csc/modulefiles/
```

File: docs/apps/elmer.md [hunk 4]
```diff
@@ -17,84 +17,115 @@ directives implemented.
...
+    module load elmer/latest
+    # make sure the SIF is in the start-info
+    echo myrun.sif > ELMERSOLVER_STARTINFO
+    echo "starting Elmer simulation with SIF file"
+    cat ELMERSOLVER_STARTINFO
+    srun ElmerSolver
+    echo "done"
+    ```    
+
+Instructions on how to submit and monitor jobs can be found [here](../computing/running/submitting-jobs.md) in CSC docs, and [here](https://docs.lumi-supercomputer.eu/runjobs/scheduled-jobs/slurm-quickstart/) in LUMI docs.
```
Comment (hdrei) [inline docs/apps/elmer.md:125]:
```suggestion
Instructions on how to submit and monitor jobs can be found in [CSC docs](../computing/running/submitting-jobs.md) and in [LUMI docs](https://docs.lumi-supercomputer.eu/runjobs/scheduled-jobs/slurm-quickstart/).
```

**Review by hdrei — COMMENTED** [review]
We checked this through with @leopekkas and the Elmer part should be fine now

**Comment by rkronberg** [comment]
Thanks @hdrei and @leopekkas! What about @ejarvine? Are the OpenFOAM changes ok?

**Review by hdrei — APPROVED** [review]

**Pull request merged**